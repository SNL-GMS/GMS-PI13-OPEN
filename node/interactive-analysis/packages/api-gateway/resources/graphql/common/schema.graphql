# A date/time type
scalar Date

# The log level
enum LogLevel {
  error
  warn
  client
  info
  timing
  data
  debug
}

# Input verion of a time range
input TimeRange {
  startTime: Float!
  endTime: Float!
}

# The version of the software, and commit SHA for displaying
type VersionInfo {
  versionNumber: String
  commitSHA: String
}

# The client log
type ClientLog {
  logLevel: LogLevel!
  message: String!
  time: String!
}

# The client log input
input ClientLogInput {
  logLevel: LogLevel!
  message: String!
  time: String
}

# The following queries are supported
type Query {
  # Query count (Dummy)
  queryCount: Int

  # The version information
  versionInfo: VersionInfo
}

# The following mutations are supported
type Mutation {
  # Mutation count (Dummy)
  mutationCount: Int

  # Client log
  clientLog(logs: [ClientLogInput]): ClientLog
}

# The following subscriptions are supported
type Subscription {
  # Subscription Count (Dummy)
  subscriptionCount: Int
}
