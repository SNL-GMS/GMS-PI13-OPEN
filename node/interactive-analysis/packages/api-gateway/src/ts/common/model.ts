import { LogLevel } from '@gms/common-graphql/lib/graphql/common/types';

/**
 * Common model definitions shared across gateway data APIs
 */

/**
 * Represents an association mapping
 * between an event and signal detection.
 */
export interface Association {
  /** the event association id */
  readonly associationId: string;
  /** the event id */
  readonly eventId: string;
  /** the event hypothesis id */
  readonly eventHypothesisId: string;
  /** the signal detection id */
  readonly signalDetectionId: string;
  /** the signal detection hypothesis id */
  readonly signalDetectionHypothesisId: string;
  /** the association reject flag */
  readonly rejected: boolean;
}

/**
 * Represents a location specified using latitude (degrees), longitude (degrees),
 * and altitude (kilometers).
 */
export interface Location {
  readonly latitudeDegrees: number;
  readonly longitudeDegrees: number;
  readonly elevationKm: number;
  readonly depthKm?: number;
}

/**
 * Represents a frequency range
 */
export interface FrequencyBand {
  readonly minFrequencyHz: number;
  readonly maxFrequencyHz: number;
}

/**
 * Time range
 */
export interface TimeRange {
  readonly startTime: number;
  readonly endTime: number;
}

/**
 * Keeps track of state between different workspaces
 */
export interface WorkspaceState {
  readonly eventToUsers: EventToUsers[];
}

/**
 * Event with list of usernames
 */
export interface EventToUsers {
  readonly eventId: string;
  readonly userNames: string[];
}

/**
 * Enumeration representing the different types of stations in the monitoring network.
 */
export enum StationType {
  Seismic3Component = 'Seismic3Component',
  Seismic1Component = 'Seismic1Component',
  SeismicArray = 'SeismicArray',
  Hydroacoustic = 'Hydroacoustic',
  HydroacousticArray = 'HydroacousticArray',
  Infrasound = 'Infrasound',
  InfrasoundArray = 'InfrasoundArray',
  Weather = 'Weather',
  UNKNOWN = 'UNKNOWN'
}

/**
 * Enumeration representing the different types of processing channels.
 */
export enum ChannelDataType {
  SEISMIC = 'SEISMIC',
  HYDROACOUSTIC = 'HYDROACOUSTIC',
  INFRASOUND = 'INFRASOUND',
  WEATHER = 'WEATHER',
  DIAGNOSTIC_SOH = 'DIAGNOSTIC_SOH',
  DIAGNOSTIC_WEATHER = 'DIAGNOSTIC_WEATHER'
}

/**
 * Processing Context part of data structure that filtered waveform
 * control service uses to tell who/where the request came from. In our case
 * from Interactive UI user
 */
export interface ProcessingContext {
  readonly analystActionReference: AnalystActionReference;
  readonly processingStepReference: ProcessingStepReference;
  readonly storageVisibility: string;
}

/**
 * Analyst action reference
 */
export interface AnalystActionReference {
  readonly processingStageIntervalId: string;
  readonly processingActivityIntervalId: string;
  readonly analystId: string;
}

/**
 * Processing step reference
 */
export interface ProcessingStepReference {
  readonly processingStageIntervalId: string;
  readonly processingSequenceIntervalId: string;
  readonly processingStepId: string;
}

/**
 * Test data paths used when reading in data
 */
export interface TestDataPaths {
  readonly dataHome: string;
  readonly jsonHome: string;
  readonly fpHome: string;
  readonly fkHome: string;
  readonly channelsHome: string;
  readonly additionalDataHome: string;
  readonly integrationDataHome: string;
  readonly tempTigerTeamData: string;
}

/**
 * Represents calibration information associated with a waveform
 */
export interface ProcessingCalibration {
  readonly factor: number;
  readonly factorError: number;
  readonly period: number;
  readonly timeShift: number;
}

/**
 * Relative Position information relative to a location
 */
export interface Position {
  readonly northDisplacementKm: number;
  readonly eastDisplacementKm: number;
  readonly verticalDisplacementKm: number;
}

/**
 * Represents the configured type of data source the API Gateway provides access to - values:
 * Local - The API gateway loads data from local file storage for testing purposes
 * Service - The API gateway uses services to provide access to backend (e.g. OSD) data
 */
export enum AccessorDataSource {
  Local = 'Local',
  Service = 'Service'
}

/**
 * Enumerated list of source types used to compute distances to
 */
export enum DistanceSourceType {
  Event = 'Event',
  UserDefined = 'UserDefined'
}

/**
 * Distance value's units degrees or kilometers
 */
export enum DistanceUnits {
  degrees = 'degrees',
  km = 'km'
}

/**
 * The distance value representing degrees and km.
 */
export interface Distance {
  readonly degrees: number;
  readonly km: number;
}

/**
 * Represents input arguments for calculating distance measurement
 * relative to a specified source location
 */
export interface DistanceToSourceInput {
  // The type of the source the distance is measured to (e.g. and event)
  readonly sourceType: DistanceSourceType;

  // the unique ID of the source object
  readonly sourceId: string;
}

/**
 * Represents a distance measurement relative to a specified source location
 */
export interface DistanceToSource {
  // The distance
  readonly distance: Distance;

  // The azimuth
  readonly azimuth: number;

  // The source location
  readonly sourceLocation: Location;

  // The type of the source the distance is measured to (e.g. and event)
  readonly sourceType: DistanceSourceType;

  // the unique ID of the source object
  readonly sourceId: string;

  // Which station distance to the source
  readonly stationId: string;
}

/**
 * Client log, object that describes the client log message
 */
export interface ClientLog {
  readonly logLevel: LogLevel;
  readonly message: string;
  readonly time?: string;
}

/**
 * Units used in DoubleValue part of feature prediction
 */
export enum Units {
  DEGREES = 'DEGREES',
  RADIANS = 'RADIANS',
  SECONDS = 'SECONDS',
  HERTZ = 'HERTZ',
  SECONDS_PER_DEGREE = 'SECONDS_PER_DEGREE',
  SECONDS_PER_RADIAN = 'SECONDS_PER_RADIAN',
  SECONDS_PER_DEGREE_SQUARED = 'SECONDS_PER_DEGREE_SQUARED',
  SECONDS_PER_KILOMETER_SQUARED = 'SECONDS_PER_KILOMETER_SQUARED',
  SECONDS_PER_KILOMETER = 'SECONDS_PER_KILOMETER',
  SECONDS_PER_KILOMETER_PER_DEGREE = 'SECONDS_PER_KILOMETER_PER_DEGREE',
  ONE_OVER_KM = 'ONE_OVER_KM',
  NANOMETERS = 'NANOMETERS',
  NANOMETERS_PER_SECOND = 'NANOMETERS_PER_SECOND',
  NANOMETERS_PER_COUNT = 'NANOMETERS_PER_COUNT',
  UNITLESS = 'UNITLESS',
  MAGNITUDE = 'MAGNITUDE',
  COUNTS_PER_NANOMETER = 'COUNTS_PER_NANOMETER',
  COUNTS_PER_PASCAL = 'COUNTS_PER_PASCAL',
  PASCALS_PER_COUNT = 'PASCALS_PER_COUNT'
}

/**
 * Phase type list
 */
export enum PhaseType {
  // TODO: need to elaborate with full set of phase labels
  P = 'P',
  S = 'S',
  P3KPbc = 'P3KPbc',
  P4KPdf_B = 'P4KPdf_B',
  P7KPbc = 'P7KPbc',
  P7KPdf_D = 'P7KPdf_D',
  PKiKP = 'PKiKP',
  PKKSab = 'PKKSab',
  PKP2bc = 'PKP2bc',
  PKP3df_B = 'PKP3df_B',
  PKSab = 'PKSab',
  PP_1 = 'PP_1',
  pPKPbc = 'pPKPbc',
  PS = 'PS',
  Rg = 'Rg',
  SKiKP = 'SKiKP',
  SKKSac = 'SKKSac',
  SKPdf = 'SKPdf',
  SKSdf = 'SKSdf',
  sPdiff = 'sPdiff',
  SS = 'SS',
  sSKSdf = 'sSKSdf',
  Lg = 'Lg',
  P3KPbc_B = 'P3KPbc_B',
  P5KPbc = 'P5KPbc',
  P7KPbc_B = 'P7KPbc_B',
  Pb = 'Pb',
  PKKP = 'PKKP',
  PKKSbc = 'PKKSbc',
  PKP2df = 'PKP2df',
  PKPab = 'PKPab',
  PKSbc = 'PKSbc',
  PP_B = 'PP_B',
  pPKPdf = 'pPKPdf',
  PS_1 = 'PS_1',
  SKKP = 'SKKP',
  SKKSac_B = 'SKKSac_B',
  SKS = 'SKS',
  SKSSKS = 'SKSSKS',
  sPKiKP = 'sPKiKP',
  SS_1 = 'SS_1',
  SSS = 'SSS',
  nNL = 'nNL',
  P3KPdf = 'P3KPdf',
  P5KPbc_B = 'P5KPbc_B',
  P7KPbc_C = 'P7KPbc_C',
  PcP = 'PcP',
  PKKPab = 'PKKPab',
  PKKSdf = 'PKKSdf',
  PKP3 = 'PKP3',
  PKPbc = 'PKPbc',
  PKSdf = 'PKSdf',
  pPdiff = 'pPdiff',
  PPP = 'PPP',
  pSdiff = 'pSdiff',
  Sb = 'Sb',
  SKKPab = 'SKKPab',
  SKKSdf = 'SKKSdf',
  SKS2 = 'SKS2',
  Sn = 'Sn',
  sPKP = 'sPKP',
  SS_B = 'SS_B',
  SSS_B = 'SSS_B',
  NP = 'NP',
  P3KPdf_B = 'P3KPdf_B',
  P5KPdf = 'P5KPdf',
  P7KPdf = 'P7KPdf',
  PcS = 'PcS',
  PKKPbc = 'PKKPbc',
  PKP = 'PKP',
  PKP3ab = 'PKP3ab',
  PKPdf = 'PKPdf',
  Pn = 'Pn',
  pPKiKP = 'pPKiKP',
  PPP_B = 'PPP_B',
  pSKS = 'pSKS',
  ScP = 'ScP',
  SKKPbc = 'SKKPbc',
  SKP = 'SKP',
  SKS2ac = 'SKS2ac',
  SnSn = 'SnSn',
  sPKPab = 'sPKPab',
  sSdiff = 'sSdiff',
  NP_1 = 'NP_1',
  P4KPbc = 'P4KPbc',
  P5KPdf_B = 'P5KPdf_B',
  P7KPdf_B = 'P7KPdf_B',
  Pdiff = 'Pdiff',
  PKKPdf = 'PKKPdf',
  PKP2 = 'PKP2',
  PKP3bc = 'PKP3bc',
  PKPPKP = 'PKPPKP',
  PnPn = 'PnPn',
  pPKP = 'pPKP',
  PPS = 'PPS',
  pSKSac = 'pSKSac',
  ScS = 'ScS',
  SKKPdf = 'SKKPdf',
  SKPab = 'SKPab',
  SKS2df = 'SKS2df',
  SP = 'SP',
  sPKPbc = 'sPKPbc',
  sSKS = 'sSKS',
  P4KPdf = 'P4KPdf',
  P5KPdf_C = 'P5KPdf_C',
  P7KPdf_C = 'P7KPdf_C',
  Pg = 'Pg',
  PKKS = 'PKKS',
  PKP2ab = 'PKP2ab',
  PKP3df = 'PKP3df',
  PKS = 'PKS',
  PP = 'PP',
  pPKPab = 'pPKPab',
  PPS_B = 'PPS_B',
  pSKSdf = 'pSKSdf',
  Sdiff = 'Sdiff',
  SKKS = 'SKKS',
  SKPbc = 'SKPbc',
  SKSac = 'SKSac',
  SP_1 = 'SP_1',
  sPKPdf = 'sPKPdf',
  sSKSac = 'sSKSac',
  Sx = 'Sx',
  tx = 'tx',
  N = 'N',
  Px = 'Px',
  PKhKP = 'PKhKP',
  UNKNOWN = 'UNKNOWN',
  LR = 'LR'
}
