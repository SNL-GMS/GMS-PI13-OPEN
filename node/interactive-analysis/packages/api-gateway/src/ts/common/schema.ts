/**
 * GraphQL schema definition for the waveform API gateway
 */
import { MILLISECONDS_IN_SECOND } from '@gms/common-util';
import { readFileSync } from 'fs';
import { GraphQLScalarType } from 'graphql';
import { Kind } from 'graphql/language';
import { resolve } from 'path';

// GraphQL schema definitions
export const schema = readFileSync(
  resolve(process.cwd(), 'resources/graphql/common/schema.graphql')
).toString();

export const extendedSchema = readFileSync(
  resolve(process.cwd(), 'resources/graphql/common/extended-schema.graphql')
).toString();

// GraphQL custom Date scalar to convert between typescript Date and seconds integer
export const gqlDate = new GraphQLScalarType({
  name: 'Date',
  description: 'Date custom scalar type',
  /**
   * parse date value as number from client
   * @param value the value
   */
  parseValue(value: number) {
    // Input is expected in seconds; convert to milliseconds to
    // satisfy the Javascript Date API
    return new Date(value * MILLISECONDS_IN_SECOND);
  },
  /**
   * Serialize Date object to milliseconds to send to client
   * @param value the value
   */
  serialize(value: Date) {
    // Output is expected in seconds; convert from milliseconds to
    // which is used in the Javascript Date API
    return value.getTime() / MILLISECONDS_IN_SECOND;
  },
  /**
   * Parse the ast node
   * @param ast - should be an INT or FLOAT
   */
  parseLiteral(ast) {
    if (ast.kind === Kind.INT) {
      const base = 10;
      return parseInt(ast.value, base);
    }
    if (ast.kind === Kind.FLOAT) {
      return parseFloat(ast.value);
    }
    return undefined;
  }
});
