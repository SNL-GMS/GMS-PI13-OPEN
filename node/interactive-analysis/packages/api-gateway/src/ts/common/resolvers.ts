import { LogLevel } from '@gms/common-graphql/lib/graphql/common/types';
import { PubSub } from 'graphql-subscriptions';
import { CacheProcessor } from '../cache/cache-processor';
import { UserContext } from '../cache/model';
import { gatewayLogger as logger } from '../log/gateway-logger';
import * as model from './model';
/**
 * Resolvers for the common API gateway
 */
const HASH_LENGTH = 8;

// GraphQL Resolvers
logger.info('Creating common API Gateway GraphQL resolvers...');
export const pubsub = new PubSub();

export const resolvers = {
  Query: {
    versionInfo: async (_, __, userContext: UserContext) => ({
      versionNumber: process.env.CI_COMMIT_REF_NAME
        ? process.env.CI_COMMIT_REF_NAME
        : process.env.GIT_BRANCH
        ? process.env.GIT_BRANCH
        : 'development',
      commitSHA: String(process.env.GIT_COMMITHASH).substr(0, HASH_LENGTH)
    })
  },

  Mutation: {
    clientLog: async (_, { logs }: { logs: model.ClientLog[] }, userContext: UserContext) => {
      new Promise((resolve, reject) => {
        logs.forEach((log: model.ClientLog) => {
          if (log.logLevel === LogLevel.timing) {
            logger.timing(log.message, userContext.userName);
          } else {
            logger.client(log.message, log.logLevel, userContext.userName);
          }
        });
        resolve();
      });
      return undefined;
    }
  }
};

export const extendedResolvers = {
  Query: {
    workspaceState: async (_, { timeRange }, userContext: UserContext) =>
      CacheProcessor.Instance().getWorkspaceState()
  }
};
