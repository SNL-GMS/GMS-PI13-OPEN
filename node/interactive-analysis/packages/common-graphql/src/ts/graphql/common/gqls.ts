import gql from 'graphql-tag';

export const locationFragment = gql`
  fragment LocationFragment on Location {
    latitudeDegrees
    longitudeDegrees
    elevationKm
  }
`;

export const workspaceStateFragment = gql`
  fragment WorkspaceStateFragment on WorkspaceState {
    eventToUsers {
      eventId
      userNames
    }
  }
`;

export const versionInfoFragment = gql`
  fragment VersionInfoFragment on VersionInfo {
    versionNumber
    commitSHA
  }
`;
