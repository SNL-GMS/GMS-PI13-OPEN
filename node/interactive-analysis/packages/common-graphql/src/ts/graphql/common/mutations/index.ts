export { clientLogMutation } from './client-log';
