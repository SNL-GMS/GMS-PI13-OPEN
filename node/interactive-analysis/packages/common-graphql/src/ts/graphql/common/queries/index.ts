export { workspaceStateQuery } from './workspace-state-query';
export { versionInfoQuery } from './version-info-query';
