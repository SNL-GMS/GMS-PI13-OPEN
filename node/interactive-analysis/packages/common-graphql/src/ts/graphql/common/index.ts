import * as CommonGqls from './gqls';
import * as CommonMutations from './mutations';
import * as CommonQueries from './queries';
import * as CommonSubscriptions from './subscriptions';
import * as CommonTypes from './types';
export { CommonGqls, CommonMutations, CommonQueries, CommonSubscriptions, CommonTypes };
