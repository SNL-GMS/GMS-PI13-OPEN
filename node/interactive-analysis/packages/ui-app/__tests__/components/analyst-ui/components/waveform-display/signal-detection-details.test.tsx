import React from 'react';
import { SignalDetectionDetails } from '../../../../../src/ts/components/analyst-ui/common/dialogs';
import { signalDetectionsData } from '../../../../__data__/signal-detections-data';

// tslint:disable-next-line:no-implicit-dependencies no-var-requires no-require-imports
const Enzyme = require('enzyme');
// set up window alert and open so we don't see errors
window.alert = jest.fn();
window.open = jest.fn();

it('SignalDetectionDetails renders & matches snapshot', () => {
  const wrapper = Enzyme.shallow(
    <SignalDetectionDetails detection={signalDetectionsData[0]} color={'red'} />
  );
  expect(wrapper).toMatchSnapshot();
});
