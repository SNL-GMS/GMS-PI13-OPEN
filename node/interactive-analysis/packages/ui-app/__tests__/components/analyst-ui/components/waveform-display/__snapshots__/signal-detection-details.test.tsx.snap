// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignalDetectionDetails renders & matches snapshot 1`] = `
<div>
  <Form
    defaultPanel={
      Object {
        "formItems": Array [
          Object {
            "displayText": "P",
            "itemKey": "Phase",
            "itemType": "Display",
            "labelText": "Phase",
          },
          Object {
            "displayText": "2010/05/20T22:07:17.850000",
            "displayTextFormat": "Time",
            "itemKey": "Detection time",
            "itemType": "Display",
            "labelText": "Detection time",
          },
          Object {
            "displayText": "0.69 s",
            "itemKey": "Time uncertainty",
            "itemType": "Display",
            "labelText": "Time uncertainty",
          },
          Object {
            "displayText": "No",
            "itemKey": "Rejected",
            "itemType": "Display",
            "labelText": "Rejected",
          },
        ],
        "key": "Current Version",
      }
    }
    disableSubmit={true}
    extraPanels={
      Array [
        Object {
          "content": <div
            className="ag-theme-dark signal-detection-details-versions-table"
          >
            <div
              className="max"
            >
              <ClientSideTable
                columnDefs={
                  Array [
                    Object {
                      "cellStyle": Object {
                        "text-align": "left",
                      },
                      "field": "phase",
                      "headerName": "Phase",
                      "width": 70,
                    },
                    Object {
                      "cellStyle": Object {
                        "text-align": "left",
                      },
                      "field": "arrivalTimeMeasurementTimestamp",
                      "headerName": "Detection time",
                      "valueFormatter": [Function],
                      "width": 165,
                    },
                    Object {
                      "cellStyle": Object {
                        "text-align": "left",
                      },
                      "field": "arrivalTimeMeasurementUncertaintySec",
                      "headerName": "Time uncertainty",
                      "valueFormatter": [Function],
                      "width": 125,
                    },
                    Object {
                      "cellStyle": Object {
                        "text-align": "left",
                      },
                      "field": "rejected",
                      "headerName": "Rejected",
                      "valueFormatter": [Function],
                      "width": 75,
                    },
                  ]
                }
                getRowNodeId={[Function]}
                id="table-signal-detection-details"
                overlayNoRowsTemplate="No Verions"
                rowClassRules={
                  Object {
                    "versions-table__row--first-in-table": [Function],
                  }
                }
                rowData={
                  Array [
                    Object {
                      "arrivalTimeMeasurementFeatureType": "ARRIVAL_TIME",
                      "arrivalTimeMeasurementTimestamp": 1274393334.1,
                      "arrivalTimeMeasurementUncertaintySec": 0.685,
                      "first-in-table": true,
                      "id": "b90ee9c8-3fc9-4ccd-95ea-5d1d206c35a7",
                      "phase": "tx",
                      "rejected": false,
                      "versionId": "b90ee9c8-3fc9-4ccd-95ea-5d1d206c35a7",
                    },
                    Object {
                      "arrivalTimeMeasurementFeatureType": "ARRIVAL_TIME",
                      "arrivalTimeMeasurementTimestamp": 1274393334.1,
                      "arrivalTimeMeasurementUncertaintySec": 0.685,
                      "id": "e4ae55cc-a395-471c-8a98-bb2cf08411cd",
                      "phase": "P",
                      "rejected": false,
                      "versionId": "e4ae55cc-a395-471c-8a98-bb2cf08411cd",
                    },
                  ]
                }
                rowSelection="single"
              />
            </div>
          </div>,
          "key": "All Versions",
        },
      ]
    }
    header="Signal Detection"
    headerDecoration={
      <div
        className="signal-detection-swatch"
        style={
          Object {
            "backgroundColor": "red",
          }
        }
      />
    }
    onCancel={[Function]}
  />
</div>
`;
